cspatients/tests.py

```diff
@@ -356,3 +446,212 @@ def test_right_patient_id_bad_dict(self):
+            )
+        self.assertTrue(PatientEntry.objects.get(patient_id="HLFSH"))
+
+
+class RPPatientExistsTest(TestCase):
+
+    def setUp(self):
+        delete_whole_table()
+        self.client = Client()
+        self.good_data = SAMPLE_RP_POST_DATA
+        self.bad_data = SAMPLE_RP_POST_DATA_2
+
+        patient = Patient.objects.create(
+            name="Jane Doe",
+            patient_id="HLFSH",
+            age=20
+        )
+
+        PatientEntry.objects.create(
+            patient_id=patient
+        )
+
+    def test_finds_existing_patient(self):
+
+        response = self.client.post(
+            "/cspatients/api/rppatientexists",
+            self.good_data
+        )
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_does_not_find_not_existing_patient(self):
+
+        response = self.client.post(
+            "/cspatients/api/rppatientexists",
+            self.bad_data
+        )
+
+        self.assertEqual(response.status_code, 404)
+
+
+class RPEntryChangesTest(TestCase):
+
+    """
+        Test the rp_entrychanges endpoint.
+    """
+    def setUp(self):
+        delete_whole_table()
+        self.client = Client()
+        self.good_data = SAMPLE_RP_POST_DATA
+        self.bad_data = SAMPLE_RP_POST_DATA_2
+
+        patient = Patient.objects.create(
+            name="Jane Doe",
+            patient_id="HLFSH",
+            age=20
+        )
+
+        PatientEntry.objects.create(
+            patient_id=patient
+        )
+
+    def test_good_change_name(self):
+
+        response = self.client.post(
+            "/cspatients/api/rpentrychanges",
+            self.good_data
+        )
+        # Test the right response code
+        self.assertEqual(response.status_code, 200)
+
+        # Test that that the change has been made
+
+        patient = Patient.objects.get(patient_id="HLFSH")
+        self.assertTrue(patient.name == "Nyasha")
+
+    def test_non_existing_entry_change_name(self):
+
+        response = self.client.post(
+            "/cspatients/api/rpentrychanges",
+            self.bad_data
+        )
+        # Test the right response code 400
+
+        self.assertEqual(response.status_code, 400)
+
+
+class RPEntryDeliveredTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.good_data = SAMPLE_RP_POST_DATA
+        self.bad_data = SAMPLE_RP_POST_DATA_2
+
+        patient = Patient.objects.create(
+            name="Jane Doe",
+            patient_id="HLFSH",
+            age=20
+        )
+
+        PatientEntry.objects.create(
+            patient_id=patient
+        )
+
+    def test_patient_who_exists_delivered(self):
+
+        response = self.client.post(
+            "/cspatients/api/rpentrydelivered",
+            self.good_data
+        )
+        # Test returns the right response code
+        self.assertEqual(response.status_code, 200)
+
+        # Test that the delivery time has been updated
+        self.assertIsNotNone(
```

Comment:
`assertIsNotNone` is quite a vague test and will pass under circumstances that you might not want. It would be better in this case to make sure that you are testing that this value has been updated to a specific value.